src/status_im2/contexts/chat/bottom_sheet_composer/view.cljs

```diff
@@ -109,42 +114,46 @@
-                 :accessibility-label      :chat-message-input}]
-               [gradients/view props state animations show-bottom-gradient?]]
-              [images/images-list]
-              [actions/view props state animations window-height insets (seq images)]]]))]))])
+            (utils/update-input props state input-text)
+            [:<>
+             [mentions/view props state animations max-height cursor-pos]
+             [gesture/gesture-detector
+              {:gesture (drag-gesture/drag-gesture props state animations dimensions keyboard-shown)}
+              [reanimated/view
+               {:style     (style/sheet-container insets state animations)
+                :on-layout #(handler/layout % state blur-height)}
+               [sub-view/bar]
+               [reply/view state]
+               [edit/view edit #(utils/cancel-edit-message state animations)]
+               [reanimated/touchable-opacity
+                {:active-opacity      1
+                 :on-press            (when @(:input-ref props) #(.focus ^js @(:input-ref props)))
+                 :style               (style/input-container (:height animations) max-height)
+                 :accessibility-label :message-input-container}
+                [rn/text-input
+                 {:ref                      #(reset! (:input-ref props) %)
+                  :default-value            @(:text-value state)
+                  :on-focus                 #(handler/focus props state animations dimensions)
+                  :on-blur                  #(handler/blur state animations dimensions images reply)
+                  :on-content-size-change   #(handler/content-size-change %
+                                                                          state
+                                                                          animations
+                                                                          dimensions
+                                                                          (or keyboard-shown edit))
+                  :on-scroll                #(handler/scroll % props state animations dimensions)
+                  :on-change-text           #(handler/change-text % props state)
+                  :on-selection-change      #(handler/selection-change % state)
+                  :max-height               max-height
+                  :max-font-size-multiplier 1
+                  :multiline                true
+                  :placeholder              (i18n/label :t/type-something)
+                  :placeholder-text-color   (colors/theme-colors colors/neutral-40 colors/neutral-50)
+                  :style                    (style/input props state)
+                  :accessibility-label      :chat-message-input}]
+                [gradients/view props state animations show-bottom-gradient?]]
+               [images/images-list]
+               [actions/view props state animations window-height insets (seq images)]]]]))]))])
```

Comment:
@OmarBasem, if possible, prefer to pass less complex data structures to components so that Reagent has less work to do to compare props passed to components. In this case, passing `(seq images)` is unnecessary, since `actions/view` expects a boolean `images?`.

> Returns true if coll has no items - same as (not (seq coll)). Please use the idiom (seq x) rather than (not (empty? x))
>
> -- Docstring of `empty?`

This is an interesting case where the docstring is misleading for Reagent development. In fact, we need to do the opposite of the Clojure recommendation and pass `(boolean (seq images))`. Note that I'm not saying the idiom `(seq images)` is generally bad, I'm just saying prop complexity should be minimized to reduce the cost of prop equality checks. The idiom `(seq images)` is fine "inside" functions.

cc'ing @flexsurfer, @erikseppanen, @ulisesmac

Reply:
Thanks @ilmotta, I will refactor it.